bowling: add to track

dawcars opened the pull request:
<!-- Your content goes here: -->
I have started work on bowling exercise. Work in progress


<!-- DO NOT EDIT BELOW THIS LINE! -->
---

Reviewer Resources:

[Track Policies](https://github.com/exercism/xjava/blob/master/POLICIES.md#event-checklist)

FridaTveit commented:
Thanks for tackling this @dawcars! :D I changed the name to "exercise-slug: dibs on implementing exercise". Once you're ready for a review feel free to change it to "exercist-slug: add to track". This is just to make it obvious to other contributors that you're working on an exercise, and obvious to reviewers once you're ready for a review :)

dawcars commented:
Thanks @FridaTveit. That's really helpful. I will change the name when it is ready for review. Thanks for your help. :)

FridaTveit commented:
No problem! :)

dawcars commented:
Please can you check this is OK to add to the track. It should now be ready. Many thanks

Smarticles101 commented on exercises/bowling/src/main/java/BowlingGame.java:
```diff
@@ -0,0 +1,11 @@
+public class BowlingGame {
```
Since the spot you have chosen to put this is >21, you can remove this file and replace it with a `.keep` file. (See the [POLICIES.md](https://github.com/exercism/java/blob/master/POLICIES.md#starter-implementations) file)

Smarticles101 commented on exercises/bowling/src/example/java/BowlingGame.java:
```diff
@@ -0,0 +1,98 @@
+    private int frameScore(int frameIndex) {
+        return rolls.get(frameIndex) + rolls.get(frameIndex + 1);
+    }
+}
```
I noticed your formatting of this file is a little inconsistent, specifically on where you put your curly braces. Some are on the same line, but some are on the next.

Smarticles101 replied:
Ah I found the problem. This seems to be more or less copied from the [csharp track](https://github.com/exercism/csharp/blob/master/exercises/bowling/Example.cs). If you could just fix up the formatting that would be great. (In IntelliJ just press Ctrl+Shift+Alt+L to reformat the code)

Smarticles101 commented on exercises/bowling/src/test/java/BowlingTest.java:
```diff
@@ -0,0 +1,268 @@
+        game = new BowlingGame();
+    }
+    
+    private void PlayGame(int[] rolls)
```
typically only constructors start with a capital letter. Instead of `PlayGame` it should be `playGame`

Smarticles101 commented on exercises/bowling/src/test/java/BowlingTest.java:
```diff
@@ -0,0 +1,268 @@
+    @Test
+    public void shouldBeAbleToScoreAGameWithNoStrikesOrSpares() {
+        int[] rolls = { 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6, 3, 6 };
+        PlayGame(rolls);
```
A newline would be nice to have in between these two lines in all the test cases

Smarticles101 commented on exercises/bowling/src/test/java/BowlingTest.java:
```diff
@@ -0,0 +1,268 @@
+    public void aSpareFollowedByZerosIsWorthTenPoints() {
+        int[] rolls = { 6, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+        PlayGame(rolls);
+        assertEquals(10, game.score());
```
It might improve readability a little bit if the expected score was a variable instead

dawcars commented:
Thanks for your comments @Smarticles101. They are very helpful. I will make corrections based on your comments. If you find anything else that needs alteration, please let me know.

stkent commented:
Now that this has entered the review phase it's fine to leave it titled as "add to track" :)

dawcars commented:
I have refactored changes as suggested by @Smarticles101 and uploaded to branch. All gradle tests are passing and I have rebased with the upstream. However I see that one of the checks is failing. Does anyone know why this might be?

Smarticles101 commented:
@dawcars I'm trying to look into why it's failing

Smarticles101 commented:
So far, I haven't been able to find out why it's failing. :confused:

FridaTveit commented:
I've restarted it to see if that helps :) Sometimes the build fails for no apparent reason... But restarting it often helps :)

stkent commented:
[If this was branched off master before https://github.com/exercism/java/pull/674 was merged, that could be the cause.]

Smarticles101 commented:
Thanks @dawcars :)

dawcars commented:
Thanks for your help with this @Smarticles101, @stkent and @FridaTveit. It's very much appreciated. :)
